lib/shrinkwrap.js

```diff
@@ -33,10 +36,32 @@ function shrinkwrap (args, silent, cb) {
   npm.config.set('production', true)
   readPackageTree(dir, andRecalculateMetadata(iferr(cb, function (tree) {
     var pkginfo = treeToShrinkwrap(tree, !!npm.config.get('dev') || /^dev(elopment)?$/.test(npm.config.get('also')))
-    shrinkwrap_(pkginfo, silent, cb)
+
+    readPackage(iferr(cb, function(data) {
+      chain([
+        [lifecycle, data, 'preshrinkwrap'],
+        [shrinkwrap_, pkginfo, silent],
+        [lifecycle, data, 'postshrinkwrap']
+      ], cb)
+    }))
   })))
 }
 
+function readPackage (cb) {
```

Comment:
There is! (The module `read-package-json` is how this is ordinarily done.) But see above!